File: resource-registry/src/main/java/nl/haploid/octowight/registry/repository/Sequence.java
package nl.haploid.octowight.registry.repository;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "sequence")
public class Sequence {

	@Id
	private String key;

	private long value;

	public String getKey() {
		return key;
	}

	public void setKey(final String key) {
		this.key = key;
	}

	public long getValue() {
		return value;
	}

	public void setValue(final long value) {
		this.value = value;
	}
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/repository/Sequence.scala
package nl.haploid.octowight.registry.repository

import java.lang

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

import scala.beans.BeanProperty

// TODO: this is a dmo
@Document(collection = "sequence")
class Sequence {

  @Id
  @BeanProperty var key: String = _
  @BeanProperty var value: lang.Long = _
}
